### PR TITLE
Split microprofile.ls project 

### PR DESCRIPTION
**Pull request by rzgry**
Part of #295 

Split microprofile.ls project into microprofile-ls (the core LS for microprofile) and quarkus-ls-ext: (SPI contributions to microprofile-ls)

### CODE REVIEW

File: quarkus.ls.ext/com.redhat.quarkus.ls/.mvn/wrapper/maven-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/.mvn/wrapper/maven-wrapper.properties:1]: that's an old version of Maven, can you update the Maven wrappers everywhere as part of this PR? Just call mvn -N io.takari:maven:0.7.7:wrapper at the root of each project containing the wrapper

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml [hunk 2]
```diff
@@ -0,0 +1,173 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>com.redhat.microprofile</groupId>
+	<artifactId>com.redhat.quarkus.ls</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+
+	<name>Quarkus Language Server</name>
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:9]: Quarkus Extension for the MicroProfile Language Server

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+				<executions>
+					<execution>
+						<id>parse-version</id>
+						<goals>
+							<goal>parse-version</goal>
+						</goals>
+					</execution>
+				</executions>
+			</plugin>
+			<plugin>
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:80]: We don't want to generate a full blown distro here. Might be more interesting to [copy all quarkus dependencies ](https://maven.apache.org/plugins/maven-dependency-plugin/copy-dependencies-mojo.html)(minus the ones already provided by the MP LS) in one directory, those dependencies will then be copied along the quarkus extension into the final server directory.

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+				</configuration>
+			</plugin>
+		</plugins>
+	</build>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.eclipse.lsp4j</groupId>
+			<artifactId>org.eclipse.lsp4j</artifactId>
+			<version>${lsp4j.version}</version>
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:131]: scope provided

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+				<exclusion>
+					<groupId>org.codehaus.mojo</groupId>
+					<artifactId>animal-sniffer-annotations</artifactId>
+				</exclusion>
+			</exclusions>
+		</dependency>
+		<dependency>
+			<groupId>org.eclipse.lsp4j</groupId>
+			<artifactId>org.eclipse.lsp4j.jsonrpc</artifactId>
+			<version>${lsp4j.version}</version>
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:146]: scope provided

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+		</dependency>
+		<dependency>
+			<groupId>com.redhat.microprofile</groupId>
+			<artifactId>com.redhat.microprofile.ls</artifactId>
+			<version>0.0.6-SNAPSHOT</version>
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:151]: scope provided

**Comment by jeffmaury** [comment]
Not sure how this will affect JBoss Tools as we are pulling the bits from a P2 repo that was produced my Maven and Jenkins

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml (hunk 3 above)
Comment (rzgry) [reply quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:80]: @fbricon updated to use the `maven-jar-plugin` + `maven-dependency-plugin`

File: .travis.yml [hunk 7]
```diff
@@ -13,7 +13,8 @@ cache:
 
 script:
 - "cd microprofile.jdt && ./mvnw clean verify"
-- "cd ../microprofile.ls/com.redhat.microprofile.ls && ./mvnw clean verify"
+- "cd ../microprofile.ls/com.redhat.microprofile.ls && ./mvnw clean verify install"
```
Comment (fbricon) [inline .travis.yml:16]: cd ../microprofile.ls/com.redhat.microprofile.ls && ./mvnw clean install

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml [hunk 8]
```diff
@@ -0,0 +1,175 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>com.redhat.microprofile</groupId>
+	<artifactId>com.redhat.quarkus.ls</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:7]: <version>0.0.8-SNAPSHOT</version>

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+		</developer>
+	</developers>
+
+	<properties>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<maven.compiler.target>1.8</maven.compiler.target>
+		<maven.compiler.source>1.8</maven.compiler.source>
+		<maven.build.timestamp.format>yyyyMMdd-HHmm</maven.build.timestamp.format>
+		<dev.build.timestamp>${maven.build.timestamp}</dev.build.timestamp>
+		<lsp4j.version>0.9.0</lsp4j.version>
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:41]: <microprofile.ls.version>${project.version}</microprofile.ls.version>

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+		<dependency>
+			<groupId>org.eclipse.lsp4j</groupId>
+			<artifactId>org.eclipse.lsp4j.jsonrpc</artifactId>
+			<version>${lsp4j.version}</version>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>com.redhat.microprofile</groupId>
+			<artifactId>com.redhat.microprofile.ls</artifactId>
+			<version>0.0.6-SNAPSHOT</version>
```
Comment (fbricon) [inline quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml:152]: `<version>${microprofile.ls.version}</version>`

File: quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml (hunk 8 above)
Comment (rzgry) [reply quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml]: Changed to use 0.8.0-SNAPSHOT instead to match mp.ls

Comment (fbricon) [reply quarkus.ls.ext/com.redhat.quarkus.ls/pom.xml]: right, that's what I meant ;-)

**Comment by fbricon** [comment]
Thanks @rzgry!

**Pull request merged**